plugins/ConnectedLivingSpace/CLSAddon.cs

```diff
@@ -272,75 +279,56 @@ private void RebuildCLSVessel(Vessel newVessel)
 
         private void RebuildCLSVessel(Part newRootPart)
         {
-            //Debug.Log("RebuildCLSVessel");
-            // Before we rebuild the vessel, we need to take some steps to tidy up the highlighting and our idea of which space is the selected space. We will make a list of all the parts that are currently in the selected space. We will also unhighlight parts that are highlighted. Once the rebuild is complete we will work out which space will be the selected space based on the first part in our list that we find in oneof the new spaces. We can then highlight that new space.
+            try
+            {
+                //Debug.Log("RebuildCLSVessel");
+                // Before we rebuild the vessel, we need to take some steps to tidy up the highlighting and our idea of which space is the selected space. We will make a list of all the parts that are currently in the selected space. We will also unhighlight parts that are highlighted. Once the rebuild is complete we will work out which space will be the selected space based on the first part in our list that we find in oneof the new spaces. We can then highlight that new space.
+                // This needs to be revised to be based on part level highlighting.. to ensure that other mod interactions are addressed.
 
-            List<uint> listSelectedParts = new List<uint>();
+                uint flightID = 0;
+                List<CLSPart> listHighlightedParts = new List<CLSPart>();

```

Comment:
Modified list to make it space agnostic.  a part.highlighted flag manages whether or not the part needs highlighting between rebuilds of the vessel.